monitoring/mock_uss/geoawareness/parsers/ed269/__init__.py

```diff
@@ -0,0 +1,151 @@
+import json
+from enum import Enum
+from typing import List, Any, Optional, Union, Dict, Tuple
+from implicitdict import ImplicitDict, StringBasedDateTime, StringBasedTimeDelta
+
+
+class Restriction(str, Enum):
+    PROHIBITED = "PROHIBITED"
+    REQ_AUTHORISATION = "REQ_AUTHORISATION"
+    CONDITIONAL = "CONDITIONAL"
+    NO_RESTRICTION = "NO_RESTRICTION"
+
+
+class Reason(str, Enum):
+    AIR_TRAFFIC = "AIR_TRAFFIC"
+    SENSITIVE = "SENSITIVE"
+    PRIVACY = "PRIVACY"
+    POPULATION = "POPULATION"
+    NATURE = "NATURE"
+    NOISE = "NOISE"
+    FOREIGN_TERRITORY = "FOREIGN_TERRITORY"
+    EMERGENCY = "EMERGENCY"
+    OTHER = "OTHER"
+
+
+class YESNO(str, Enum):
+    YES = "YES"
+    NO = "NO"
+
+
+class Purpose(str, Enum):
+    AUTHORIZATION = "AUTHORIZATION"
+    NOTIFICATION = "NOTIFICATION"
+    INFORMATION = "INFORMATION"
+
+
+class UASZoneAuthority(ImplicitDict):
+    name: Optional[str]  # max length: 200
+    service: Optional[str]  # max length: 200
+    email: Optional[str]
+    contactName: Optional[str]  # max length: 200
+    siteURL: Optional[str]
+    phone: Optional[str]  # max length: 200
+    purpose: Optional[Purpose]
+    intervalBefore: Optional[str]
+
+
+class VerticalReferenceType(str, Enum):
+    AGL = "AGL"
+    AMSL = "AMSL"
+
+
+# TODO: Discuss with @ben about Union[PolygonType, CircleType] which throws a not implemented exception
+#  in implicitdict.
+#
+# class PolygonType(ImplicitDict):
+#     type: str = "Polygon"
+#     coordinates: List[ # min 4 items
+#         List[ # 2 items
+#             float
+#         ]
+#     ]
+#
+#
+# class CircleType(ImplicitDict):
+#     type: str = "Circle"
+#     center: List[float] # 2 items
+#     radius: float # > 0
+
+# Start workaround
+class UniversalHorizontalProjectionType(str, Enum):
+    Circle = "Circle"
+    Polygon = "Polygon"
+
+
+class UniversalHorizontalProjection(ImplicitDict):
+    type: str
+    center: Optional[List[float]]  # 2 items
+    radius: Optional[float]  # > 0
+    coordinates: Optional[List[List[float]]]  # min 4 items  # 2 items
+
+
+# End workaround
+
+
+class UomDimensions(str, Enum):
+    M = "M"
+    FT = "FT"
+
+
+class UASZoneAirspaceVolume(ImplicitDict):
+    uomDimensions: UomDimensions
+    lowerLimit: Optional[int]
+    lowerVerticalReference: VerticalReferenceType
+    upperLimit: Optional[int]
+    upperVerticalReference: VerticalReferenceType
+    horizontalProjection: UniversalHorizontalProjection  # TODO: Implement Union[PolygonType, CircleType] in implicitdict
+
+
+class WeekDateType(str, Enum):
+    MON = "MON"
+    TUE = "TUE"
+    WED = "WED"
+    THU = "THU"
+    FRI = "FRI"
+    SAT = "SAT"
+    SUN = "SUN"
+    ANY = "ANY"
+
+
+class DailyPeriod(ImplicitDict):
+    day: List[WeekDateType]  # min items: 1, max items: 7
+    startTime: str  # TODO: Convert to a Time representation
```

Comment:
`implicitdict.StringBasedDateTime` is a great solution here if the field is an RFC-3339 string.  It would look something like this (not checked for full accuracy):

```python
from implicitdict import StringBasedDateTime

class DailyPeriod(ImplicitDict):
    startTime: StringBasedDateTime

p = DailyPeriod(startTime=StringBasedDateTime(datetime.utcnow()))
p2 = ImplicitDict.parse(json.loads(json.dumps(p)), DailyPeriod)
assert p2.startTime.datetime == p.startTime.datetime
```

Reply:
Unfortunately, startTime is only the time part of a ISO8601 datetime including the timezone (eg: `01:02:03.00Z`). Arrow do not support ISO8601 time only values.

Reply:
Correction, arrow is able to handle it. I will propose a new datastructure to the implicitdict repository. Let me know if you see a better approach.

Reply:
Correction, arrow can parse the values but uses the minimum date `1900-01-01` as reference which can lead to a date out while manipulating timezones. Based on your input, I came up with a proposal. May I kindly ask you to check the last commit?